### PR TITLE
Revert "[chore] delete unavailable S3 & Kafka Catalogs"

### PR DESCRIPTION
**Pull request by EricJoy2048**
Reverts apache/incubator-seatunnel#4477

### CODE REVIEW

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — DISMISSED** [review]

**Comment by TyrantLucifer** [comment]
Wait CI/CD

**Comment by liugddx** [comment]
Why does the sonar bot appear randomly?

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/catalog/KafkaDataTypeConvertor.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    @Override
+    public SeaTunnelDataType<?> toConnectorType(
+            SeaTunnelDataType<?> seaTunnelDataType, Map<String, Object> dataTypeProperties)
+            throws DataTypeConvertException {
+        return seaTunnelDataType;
+    }
+
+    @Override
+    public String getIdentity() {
+        return "Kafka";
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/catalog/KafkaDataTypeConvertor.java:70]: Maybe we need a constant class to manage these names uniformly.

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/catalog/KafkaCatalog.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                    bootstrapServers,
+                    topicDescription.name());
+        } catch (DatabaseNotExistException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new CatalogException(
+                    String.format(
+                            "Catalog : %s establish connection to %s error",
+                            catalogName, bootstrapServers),
+                    e);
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/catalog/KafkaCatalog.java:102]: Can throw Exception directly?

**Comment by liugddx** [comment]
Does the database and tables in the Kafka catalog refer to topics?

**Comment by ic4y** [comment]
@EricJoy2048 Does this pr still need to be merged

**Pull request closed without merging**